src/libs/actions/Link.js

```diff
@@ -49,8 +51,13 @@ function openOldDotLink(url) {
 
         const params = _.compact([authTokenParam, emailParam]).join('&');
 
-        // If the URL contains # or ?, we can assume they don't need to have the `?` token to start listing url parameters.
-        return `${CONFIG.EXPENSIFY.EXPENSIFY_URL}${url}${hasHashParams || hasURLParams ? '&' : '?'}${params}`;
+        return Environment.getOldDotEnvironmentURL()
+            .then((environmentURL) => {
+                const oldDotDomain = Url.addTrailingForwardSlash(environmentURL || CONFIG.EXPENSIFY.EXPENSIFY_URL);
+
+                // If the URL contains # or ?, we can assume they don't need to have the `?` token to start listing url parameters.
+                return `${oldDotDomain}${url}${hasHashParams || hasURLParams ? '&' : '?'}${params}`;
```

Comment:
> If the URL contains # or ?

This is wrong actually. It should be "If the URL contains **?**" (only ?)
Just because a URL has a `#` hash, it doesn't mean it has query parameters. e.g. 
`[https://](https://github.com/Expensify/App/pull/14944/files#diff` 

If we want to add query parameters to the above URL we need to start with a `?` like so
`[https://](https://github.com/Expensify/App/pull/14944/files#diff?param=1&param=2` 

There's a more straightforward way to add parameters if we build the url like so 

```js
const oldDotUrl = new URL(`${oldDotDomain}${url}`);
if (url.search) url.search += `&${params}`
else url.search = `?${params}` 

return url.toString();
```

Reply:
Yes, good catch. I'll update it.

Reply:
Wait, we have some cases where this is true. Please see this [PR](https://github.com/Expensify/App/pull/9450) where this was introduced.
So, i won't be updating now.

Reply:
Yes, thanks!